apps/api/src/schema/lib/year.ts

```diff
@@ -0,0 +1,22 @@
+import { z } from "@hono/zod-openapi";
+
+/**
+ * Confirms string for year is in valid format
+ * Must be 4 characters
+ * Could be String/Integer upon confirmation
+ *
+ * 1. Checks if it is a string
+ * 2. Checks length of string is 4
+ * 3. Checks if it is an integer
+ *
+ * Leaving a default openapi schema, if other schemas are adding to it, leave that separate.
+ *
+ */
+
+export const yearSchema = z
+  .string({ message: "Parameter 'year' is required" })
```

Comment:
Did you actually intend this message to trigger on both an absent and an incorrectly typed input? Remember that, for example, on GraphQL, an incoming `Int!` would fail to validate. Then implying the year is missing would be incorrect!